### PR TITLE
Migrate to the new resources API

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/components/AsyncImage.kt [hunk 1]
```diff
@@ -14,22 +14,22 @@ import androidx.compose.ui.Modifier
...
 
-@OptIn(ExperimentalResourceApi::class)
 @Composable
 fun AsyncImage(
     modifier: Modifier,
     imageUrl: String,
     contentDescription: String,
     contentScale: ContentScale = ContentScale.Crop,
 ) {
+    remember {  }
```
Comment (terrakok) [inline shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/components/AsyncImage.kt:28]: outdated

File: shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/components/Tag.kt [hunk 2]
```diff
@@ -46,17 +56,20 @@ fun Tag(
     }
 }
 
+@OptIn(ExperimentalResourceApi::class)
 @Composable
 fun LightningTalk(title: String, dimmed: Boolean = false) {
-    Tag(Icons.LIGHT, title, dimmed)
+    Tag(Res.drawable.light.painter(), title, dimmed)
```
Comment (terrakok) [inline shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/components/Tag.kt:62]: here and in other places I see you use a painter as a function parameter.
I would say that to pass a drawable is better and get painter in the function inside

File: shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/AboutConf.kt [hunk 3]
```diff
@@ -245,7 +247,7 @@ private fun Labs() {
 
         Row(Modifier.padding(16.dp)) {
             Icon(
-                painter = Icons.AWS_LAB,
+                painter = painterResource(Res.drawable.aws_lab),
```
Comment (terrakok) [inline shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/AboutConf.kt:250]: to be equal with other code I'd expect the `.painter()` here

**Review by terrakok — CHANGES_REQUESTED** [review]

File: shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/components/AsyncImage.kt (hunk 1 above)
Comment (e5l) [reply shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/components/AsyncImage.kt:28]: thank you, fixed

File: shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/components/Tag.kt (hunk 2 above)
Comment (e5l) [reply shared/src/mobileMain/kotlin/org/jetbrains/kotlinconf/ui/components/Tag.kt:62]: refactored

File: gradle.properties [hunk 4]
```diff
@@ -15,3 +15,4 @@ development=true
 kotlin.mpp.stability.nowarn=true
 kotlin.mpp.enableCInteropCommonization=true
 kotlin.mpp.androidSourceSetLayoutVersion=2
+compose.resources.always.generate.accessors=true
```
Comment (terrakok) [inline gradle.properties:18]: this is not required here. the plugin will generate Res class automatically

Comment (e5l) [reply gradle.properties:18]: will drop, thank you

**Pull request merged**